natlas-agent/Dockerfile

```diff
@@ -28,7 +28,7 @@ FROM ubuntu:18.04
 RUN export DEBIAN_FRONTEND=noninteractive \
 	&& apt-get update \
 	&& apt-get install --no-install-recommends -qy python3 python3-pkg-resources chromium-browser nmap xvfb vncsnapshot libcap2-bin \
-	&& setcap cap_net_raw,cap_net_admin,cap_net_bind_service+eip /usr/bin/nmap \
+	&& setcap cap_net_raw+eip /usr/bin/nmap \
```

Comment:
How certain are you of this? All documentation that I've seen has suggested the previous. https://secwiki.org/w/Running_nmap_as_an_unprivileged_user

Reply:
In docker it got all jacked up and wouldnt start (operation not permitted) without switching it to cap_net_raw+eip.  I went partially off this article after trial and error: https://raesene.github.io/blog/2017/08/27/Linux-capabilities-and-when-to-drop-all/. It could be that net_admin and net_bind just need to be allowed at the container level. 

https://docs.docker.com/engine/security/security/

Reply:
Yeah, you need to provide the caps at both the container level and at the nmap process level. Unfortunately @ajacques didn't write documentation for running the containers. According to https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities it seems like NET_RAW and NET_BIND_SERVICE are allowed by default, but NET_ADMIN is not. This might be why it was breaking. I'm not 100% sure on why nmap might need to net bind or what it needs net admin to do. I'll follow up with an nmap developer to see if they can point to the features that need these caps, or if they are just something that someone did once and it worked so they told everyone else to do it that way and never tested if it was least privilege or not.

Reply:
Ill test it that way real quick.

Reply:
``` 
docker run -it --rm --cap-add NET_ADMIN ubuntu bash
root@9a0f91593823:/# apt update
Get:1 http://security.ubuntu.com/
...
All packages are up to date.

root@9a0f91593823:/# apt install -y nmap 
Reading package lists... Done
...
Setting up nmap (7.60-1ubuntu5) ...
Processing triggers for libc-bin (2.27-3ubuntu1) ...
root@9a0f91593823:/# setcap cap_net_raw,cap_net_admin,cap_net_bind_service+eip /usr/bin/nmap
bash: setcap: command not found
root@9a0f91593823:/# apt install -y libcap2-bin
...
root@9a0f91593823:/# setcap cap_net_raw,cap_net_admin,cap_net_bind_service+eip /usr/bin/nmap
root@9a0f91593823:/# nmap
Nmap 7.60 ( https://nmap.org )
Usage: nmap [Scan Type(s)] [Options] {target specification}
SEE THE MAN PAGE (https://nmap.org/book/man.html) FOR MORE OPTIONS AND EXAMPLES
root@9a0f91593823:/# 
```